File: app/src/main/java/bassamalim/hidaya/adapters/QuranSearcherAdapter.java
package bassamalim.hidaya.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import bassamalim.hidaya.R;
import bassamalim.hidaya.activities.QuranActivity;
import bassamalim.hidaya.models.Ayah;
import bassamalim.hidaya.other.Utils;

public class QuranSearcherAdapter extends RecyclerView.Adapter<QuranSearcherAdapter.ViewHolder> {

    private final Context context;
    private final List<Ayah> suraCards;
    private final String suraStr;
    private final String tafseerString;
    private final String suraNumString;
    private final String pageNumString;
    private final String ayaNumString;

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView suraNumTv;
        private final TextView suraNameTv;
        private final TextView pageNumTv;
        private final TextView ayaNumTv;
        private final TextView ayaTextTv;
        private final TextView ayaTafseerTv;
        private final Button gotoPageBtn;

        public ViewHolder(View view) {
            super(view);
            suraNumTv = view.findViewById(R.id.sura_num_tv);
            suraNameTv = view.findViewById(R.id.sura_name_tv);
            pageNumTv = view.findViewById(R.id.page_num_tv);
            ayaNumTv = view.findViewById(R.id.aya_num_tv);
            ayaTextTv = view.findViewById(R.id.aya_text_tv);
            ayaTafseerTv = view.findViewById(R.id.aya_tafseer_tv);
            gotoPageBtn = view.findViewById(R.id.goto_page);
        }
    }

    public QuranSearcherAdapter(Context context, List<Ayah> results) {
        this.context = context;

        suraCards = results;
        suraStr = context.getString(R.string.sura);
        suraNumString = context.getString(R.string.sura_number);
        pageNumString = context.getString(R.string.page_number);
        ayaNumString = context.getString(R.string.aya_number);
        tafseerString = context.getString(R.string.tafseer);
    }

    @NonNull @Override
    public QuranSearcherAdapter.ViewHolder onCreateViewHolder(
            @NonNull ViewGroup viewGroup, int viewType) {
        return new QuranSearcherAdapter.ViewHolder(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_quran_searcher, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(QuranSearcherAdapter.ViewHolder viewHolder, final int position) {
        Ayah card = suraCards.get(position);

        String suraNumStr = suraNumString + " " + Utils.translateNumbers(
                context, String.valueOf(card.getSurah()));
        viewHolder.suraNumTv.setText(suraNumStr);

        String suraNameStr = suraStr +  " " + card.getSurahName();
        viewHolder.suraNameTv.setText(suraNameStr);

        String pageNumStr = pageNumString + " " + Utils.translateNumbers(
                context, String.valueOf(card.getPageNum()));
        viewHolder.pageNumTv.setText(pageNumStr);

        String ayaNumStr = ayaNumString + " " + Utils.translateNumbers(
                context, String.valueOf(card.getAyah()));
        viewHolder.ayaNumTv.setText(ayaNumStr);

        viewHolder.ayaTextTv.setText(card.getSS());

        String ayaTafseerStr = tafseerString + ": " + card.getTafseer();
        viewHolder.ayaTafseerTv.setText(ayaTafseerStr);

        viewHolder.gotoPageBtn.setOnClickListener(v -> {
            Intent intent = new Intent(context, QuranActivity.class);
            intent.setAction("by_page");
            intent.putExtra("page", card.getPageNum());
            context.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return suraCards.size();
    }
}

File: app/src/main/java/bassamalim/hidaya/adapters/QuranSearcherAdapter.kt
package bassamalim.hidaya.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import bassamalim.hidaya.R
import bassamalim.hidaya.activities.QuranViewer
import bassamalim.hidaya.models.Ayah
import bassamalim.hidaya.utils.LangUtils

class QuranSearcherAdapter(private val context: Context, private val items: List<Ayah>) :
    RecyclerView.Adapter<QuranSearcherAdapter.ViewHolder>() {

    private val suraStr: String = context.getString(R.string.sura)
    private val tafseerString: String = context.getString(R.string.tafseer)
    private val pageNumString: String = context.getString(R.string.page_number)
    private val ayaNumString: String = context.getString(R.string.aya_number)

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val suraNameTv: TextView
        val pageNumTv: TextView
        val ayaNumTv: TextView
        val ayaTextTv: TextView
        val ayaTafseerTv: TextView
        val gotoPageBtn: Button

        init {
            suraNameTv = view.findViewById(R.id.sura_name_tv)
            pageNumTv = view.findViewById(R.id.page_num_tv)
            ayaNumTv = view.findViewById(R.id.aya_num_tv)
            ayaTextTv = view.findViewById(R.id.aya_text_tv)
            ayaTafseerTv = view.findViewById(R.id.aya_tafseer_tv)
            gotoPageBtn = view.findViewById(R.id.goto_page)
        }
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.item_quran_searcher, viewGroup, false)
        )
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        val card = items[position]

        val suraNameStr = suraStr + " " + card.getSurahName()
        viewHolder.suraNameTv.text = suraNameStr

        val pageNumStr = "$pageNumString " +
                LangUtils.translateNums(context, card.getPageNum().toString(), false)
        viewHolder.pageNumTv.text = pageNumStr

        val ayaNumStr = "$ayaNumString " +
                LangUtils.translateNums(context, card.getAyahNum().toString(), false)
        viewHolder.ayaNumTv.text = ayaNumStr

        viewHolder.ayaTextTv.text = card.getSS()

        val ayaTafseerStr = tafseerString + ": " + card.getTafseer()
        viewHolder.ayaTafseerTv.text = ayaTafseerStr

        viewHolder.gotoPageBtn.setOnClickListener {
            val intent = Intent(context, QuranViewer::class.java)
            intent.action = "by_page"
            intent.putExtra("page", card.getPageNum())
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return items.size
    }

}